src/backends/fvm_multicore.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "catalogue_multicore.hpp"
+#include "matrix_multicore.hpp"
+
+namespace nest {
+namespace mc {
+namespace multicore {
+
+struct fvm_policy : public memory_traits {
```

Comment:
I have moved everything into one monolithic backend class for each backend. They aren't actually so big, with all the GPU-specific backend implementation taking less than 250 LOC.

The correct generic abstraction is difficult to know before we try to implement another solver. Something like FD would look very similar, and I prefer to wait until such time as we try to implement a new solver before worrying too much about making it generic.